cpp/include/cudf/fixed_point/fixed_point.hpp

```diff
@@ -0,0 +1,299 @@
+
+    auto const rhsv  = lhs._scale > rhs._scale ? detail::shift<Rad1>(rhs._value, scale_type{lhs._scale - rhs._scale}) : rhs._value;
+    auto const lhsv  = lhs._scale < rhs._scale ? detail::shift<Rad1>(lhs._value, scale_type{rhs._scale - lhs._scale}) : lhs._value;
+    auto const scale = lhs._scale > rhs._scale ? lhs._scale : rhs._scale;
+
+    #if defined(__CUDACC_DEBUG__)
+
+    assert(!addition_overflow<Rep1>(lhsv, rhsv) &&
+        "fixed_point overflow of underlying represenation type " + print_rep<Rep1>());
+
+    #endif
+
+    return fixed_point<Rep1, Rad1>{scaled_integer<Rep1>(lhsv + rhsv, scale)};
+}
+
+// MINUS Operation
+template<typename Rep1, Radix Rad1>
+fixed_point<Rep1, Rad1> operator-(fixed_point<Rep1, Rad1> const& lhs,
+                                  fixed_point<Rep1, Rad1> const& rhs) {
+
+    auto const rhsv  = lhs._scale > rhs._scale ? detail::shift<Rad1>(rhs._value, scale_type{lhs._scale - rhs._scale}) : rhs._value;
+    auto const lhsv  = lhs._scale < rhs._scale ? detail::shift<Rad1>(lhs._value, scale_type{rhs._scale - lhs._scale}) : lhs._value;
+    auto const scale = lhs._scale > rhs._scale ? lhs._scale : rhs._scale;
+
+    #if defined(__CUDACC_DEBUG__)
+
+    assert(!subtraction_overflow<Rep1>(lhsv, rhsv) &&
+        "fixed_point overflow of underlying represenation type " + print_rep<Rep1>());
+
+    #endif
+
+    return fixed_point<Rep1, Rad1>{scaled_integer<Rep1>(lhsv - rhsv, scale)};
+}
+
+// MULTIPLIES Operation
+template<typename Rep1, Radix Rad1>
+fixed_point<Rep1, Rad1> operator*(fixed_point<Rep1, Rad1> const& lhs,
+                                  fixed_point<Rep1, Rad1> const& rhs) {
+
+    #if defined(__CUDACC_DEBUG__)
+
+    assert(!multiplication_overflow<Rep1>(lhs._value, rhs._value) &&
+        "fixed_point overflow of underlying represenation type " + print_rep<Rep1>());
+
+    #endif
+
+    return fixed_point<Rep1, Rad1>{scaled_integer<Rep1>(lhs._value * rhs._value, scale_type{lhs._scale + rhs._scale})};
+}
+
+// DIVISION Operation
+template<typename Rep1, Radix Rad1>
+fixed_point<Rep1, Rad1> operator/(fixed_point<Rep1, Rad1> const& lhs,
+                                  fixed_point<Rep1, Rad1> const& rhs) {
+
+    #if defined(__CUDACC_DEBUG__)
+
+    assert(!division_overflow<Rep1>(lhs._value, rhs._value) &&
+        "fixed_point overflow of underlying represenation type " + print_rep<Rep1>());
+
+    #endif
+
+    return fixed_point<Rep1, Rad1>{scaled_integer<Rep1>(lhs._value / rhs._value, scale_type{lhs._scale - rhs._scale})};
+}
+
+// EQUALITY COMPARISON Operation
+template<typename Rep1, Radix Rad1>
+bool operator==(fixed_point<Rep1, Rad1> const& lhs,
+                fixed_point<Rep1, Rad1> const& rhs) {
+    auto const delta = std::fabs(lhs.get() - rhs.get());
+    return delta < std::numeric_limits<decltype(delta)>::epsilon();
+}
+
+template <typename Rep, Radix Radix>
+std::ostream& operator<<(std::ostream& os, fixed_point<Rep, Radix> const& si) {
```

Comment:
What does the name `si` indicate?

Reply:
Was for `scale_integer` when that was what it was originally called. Updated.